### PR TITLE
Rewrite the documentation

### PR DESCRIPTION
**Pull request by mhauru**
I rewrote large parts of the READMEs and added much new stuff. It's not done done, this is just the first pass, but I think would be good to get comments and then get this merged, and we can keep on building on this together.

One question @EdwinB12  and I talked about that we need to decide is whether we should have multiple READMEs for the many subparts of the repo or one massive one with a lot of headings. I went with the former, but actually thought the latter might be better in the end, although Edwin then did some work trying to convince me otherwise.

### CODE REVIEW

File: dtbase/backend/README.md
Comment (EdwinB12) [inline dtbase/backend/README.md:9]: Lets try and reduce this down to a single run.sh file when I merge the local installation.

File: dtbase/ingress/README.md
Comment (EdwinB12) [inline dtbase/ingress/README.md:130]:
```suggestion
weather_ingress(dt_user_email=blahblah@email.com,
    dt_user_password="this is a very bad password", 
    dt_from=dt_from, 
    dt_to=dt_to)
```

File: dtbase/backend/README.md
Comment (EdwinB12) [inline dtbase/backend/README.md:54]: TODO here that we should complete before merging for completeness

**Review by EdwinB12 — CHANGES_REQUESTED** [review]
Fabulous stuff. Some small changes and suggestions but generally fabulous! 

Now, don't hate me, but I want to reopen the discussion about putting this into a single docs readme. Think you originally argued this and having seen it laid out, I agree.

**Comment by EdwinB12** [comment]
Oh and lets add the simple ingress example in this PR.

File: README.md
Comment (EdwinB12) [inline README.md:29]: Lets make headings like this a ### header

File: DeveloperDocs.md
Comment (mhauru) [reply DeveloperDocs.md]: Done

File: dtbase/backend/README.md
Comment (mhauru) [reply dtbase/backend/README.md:62]: Done.

File: dtbase/ingress/README.md
Comment (mhauru) [reply dtbase/ingress/README.md:130]: Done

File: README.md
Comment (mhauru) [reply README.md:29]: Done

File: README.md
Comment (mhauru) [reply README.md:4]: I actually prefer the non-question version. Do you have a particular reason for preferring the (rhetorical) question one?

File: dtbase/backend/README.md
Comment (mhauru) [reply dtbase/backend/README.md:58]: I still expanded on this a bit more, now done

File: dtbase/backend/README.md
Comment (mhauru) [reply dtbase/backend/README.md:54]: I added a couple of paragraphs here. There are no more TODO notes in the docs.

**Comment by mhauru** [comment]
@EdwinB12, I attended to all the points you raised (except for one, see above), and also did a read through of everything and made lots of small changes. Could you take another look please? I'm happy to merge once you are.

**Review by EdwinB12 — APPROVED** [review]

**Pull request merged**
